### PR TITLE
Create schema for the Database

### PR DESCRIPTION
**Pull request by Esmatullah12**
In this project, my coding partner and I collaborated to develop a database schema aligned with the provided diagram.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+CREATE TABLE invoice_items(
+  id SERIAL PRIMARY KEY,
+  unit_price DECIMAL,
+  quantity INTEGER,
+  total_price DECIMAL,
+  invoice_id INTEGER,
+  treatment_id INTEGER,
+  FOREIGN KEY (invoice_id) REFERENCES invoices(id),
+  FOREIGN KEY (treatment_id) REFERENCES treatments(id)
+);
```
Comment (DeliceLydia) [inline schema_based_on_diagram.sql]: -  [ ] Please create a table between `medical_histories` and `treatments`
```
  CREATE TABLE medical_histories_has_treatments (
    medical_history_id int references medical_histories(id),
    treatment_id int references treatments(id),
    );
```

Comment (DeliceLydia) [reply schema_based_on_diagram.sql]: - [ ] Please create indexes on all FOREIGN KEYS 
ex:
```
CREATE INDEX ON medical_histories (patient_id);
```

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
### Hi Team, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- All FOREIGN KEYS are set
- All tables have a PRIMARY KEY set
- Followed git workflow

## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+CREATE TABLE medical_histories_has_treatment(
+  medical_history_id INTEGER, 
+  treatment_id INTEGER, 
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id), 
+  FOREIGN KEY (treatment_id) REFERENCES treatments(id)
```
Comment (shahadat3669) [inline schema_based_on_diagram.sql:47]: Great progress so far! You've done a lot of work. However, there are a couple of adjustments needed:

- [ ] First, you should create indexes for all your foreign keys as recommended by the previous reviewer. Currently, there are no indexes created for these two foreign keys. Please consider updating your code to meet the project requirements by adding indexes for all your foreign keys.

- [ ] Second, there seems to be a typo error. Your table name is `teatments`, but you referred to it as `treatments` in your code. You need to either change your table name to match your references or update the references to ensure there's no conflict. 

Making these changes will help ensure your database aligns with the project requirements and that there are no errors due to typos. Keep up the good work! 👍🗃️

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi everyone, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ Worked on the previous reviewer comments


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Efiamotu-1 — APPROVED** [review]
Hi @Esmatullah12 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# STATUS: APPROVED 🕺🥳 

image

## Highlights :dart: 
* Correct Gitflow :heavy_check_mark: 
* Good commit messages :heavy_check_mark: 
* Good PR summary :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, you can now Merge since it is Approved ✅, However if there are any conflicts please resolve them before merging.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**